staging/src/k8s.io/kubectl/pkg/cmd/set/helper_test.go

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package set
+
+import (
+	"reflect"
+	"testing"
+
+	v1 "k8s.io/api/core/v1"
+)
+
+func Test_updateEnv(t *testing.T) {
+	var (
+		env1 = v1.EnvVar{
+			Name:  "env1",
+			Value: "env1",
+		}
+		env2 = v1.EnvVar{
+			Name:  "env2",
+			Value: "env2",
+		}
+		env3 = v1.EnvVar{
+			Name:  "env3",
+			Value: "env3",
+		}
+	)
+
+	type args struct {
+		existing []v1.EnvVar
+		add      []v1.EnvVar
+		remove   []string
+	}
+	tests := []struct {
+		name string
+		args args
+		want []v1.EnvVar
+	}{
+		{
+			name: "case 1",
```

Comment:
Since these new tests are intended to document behavior, I'd suggest describing the behavior each case is proving in its name. I've provided some examples of what I mean as suggestions.